### PR TITLE
Acquire problemSeed from webworkJWT

### PR DESCRIPTION
**Pull request by Miniland1333**
Modified webworkJWT handling slightly so that it extracts $inputs_ref{problemSeed} into $problemSeed.

### CODE REVIEW

**Comment by Miniland1333** [comment]
As a consequence, should we make it so that `my $problemSeed  = $inputs_ref{problemSeed}//'';`?

File: lib/WeBWorK/ContentGenerator/renderViaXMLRPC.pm [hunk 1]
```diff
@@ -152,44 +152,63 @@ sub pre_header_initialize {
...
 				  CGI::li(CGI::escapeHTML([
 					"webworkJWT: |$webwork_jwt|",
-					"userID: |$user_id|", 
-					"courseID: |$courseName|",	
-					"sourceFilePath: |$sourceFilePath|",
-					"displayMode: |$displayMode|", 
-					"problemSeed: |$problemSeed|",
+					"userID: |$inputs_ref{userID}|",
+					"courseID: |$inputs_ref{courseID}|",
+					#"sourceFilePath: |$inputs_ref{sourceFilePath}|",
```
Comment (Miniland1333) [inline lib/WeBWorK/ContentGenerator/renderViaXMLRPC.pm:191]: There is a bug here where the HTML doesn't render if sourceFilePath or displayMode are uncommented

File: lib/WeBWorK/ContentGenerator/renderViaXMLRPC.pm [hunk 2]
```diff
@@ -152,44 +152,63 @@ sub pre_header_initialize {
...
 		my $sourceFilePath = $inputs_ref{sourceFilePath}//'';
-		my $payload = decode_jwt(token=>$webwork_jwt, key=>'s1r1b1r1');
+		my $payload = decode_jwt(token=>$webwork_jwt, key=>'s1r1b1r1', accepted_alg=>'HS256'); # TODO REMOVE INSECURE DEVELOPMENT KEY
+		#TODO add validation of expiration (exp), issue time (iat), not before (nbf), issuer (iss), and audience (aud).
+		# verify_exp=>1, verify_iat=>1, verify_nbf=>1, verify_exp=>1, verify_aud=>"webwork", verify_iss=>""
+		#TODO switch to asymmetric keys and JWT encrpytion [JSON Web Encryption (JWE)].
+
+
+		#override inputs_ref if keys are present
+		$inputs_ref{userID} = $payload->{userID} if $payload->{userID};
```
Comment (Miniland1333) [inline lib/WeBWorK/ContentGenerator/renderViaXMLRPC.pm:167]: We may want to make this a for loop rather than hardcoding these relations.

**Comment by Miniland1333** [comment]
Will likely still need to modify libretexts_format.pl and AttemptsTable.pm to staple the original JWT to the answerJWT

**Comment by mgage** [comment]
I'll merge this and then add the curl command to output the JWT to Eric's Adapt server.  Then I make some additional cleanup (add forloop, etc.)

**Pull request merged**